gammapy/datasets/map.py

```diff
@@ -2510,13 +2509,31 @@ def update(self, exposure, psf, edisp, geom):
         # TODO: simplify and clean up
         log.debug("Updating model evaluator")
         # cache current position of the model component
+        geom2d = geom.to_image()
+        if (
+            self.model.position is not None
+            and isinstance(geom2d, WcsGeom)
+            and not geom2d.contains(self.model.position)[0]
+        ):
+            coords = SkyCoord(
```

Comment:
You can directly use `geom.to_image().get_coord().skycoord`. As this is a very common thing to do, we could maybe also introduce `Geom.get_skycoord()` as a short-cut...